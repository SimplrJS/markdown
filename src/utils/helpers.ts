export namespace Helpers {
    export function FillArray(arr: string[], length: number, element: any = ""): string[] {
        const count = length - arr.length;

        if (count > 0) {
            const result = [...arr];
            for (let i = 0; i < count; i++) {
                result.push(element);
            }

            return result;
        } else {
            return arr;
        }
    }
}
