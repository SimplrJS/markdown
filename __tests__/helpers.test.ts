import { Helpers } from "../src/utils/helpers";

describe("FillArray", () => {
    it("fill array with empty string", () => {
        const arr = [];
        const result = Helpers.FillArray(arr, 10, "");
        expect(result).toMatchSnapshot();
    });

    it("fill array with zeroes", () => {
        const arr: number[] = [];
        const result = Helpers.FillArray(arr, 10, 0);
        expect(result).toMatchSnapshot();
    });

    it("return the same array if the size is sufficient", () => {
        const arr = ["", "", ""];
        const result = Helpers.FillArray(arr, 3, "");
        expect(result).toBe(arr);
    });
});
