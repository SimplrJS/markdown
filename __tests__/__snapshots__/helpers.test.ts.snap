// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FillArray fill array with empty string 1`] = `
Array [
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
  "",
]
`;

exports[`FillArray fill array with zeroes 1`] = `
Array [
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
]
`;
